app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -526,6 +526,10 @@ def build_toolbar_hide_button(id)
     return true if @record.kind_of?(OrchestrationStack) && @display == "instances" &&
                    %w(instance_check_compliance instance_compare).include?(id)
 
+    # hide select power management buttons for Openstack::InfraManager
+    return true if %w(host_standby host_shutdown host_reboot host_reset).include?(id) &&
+                   @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
```

Comment:
@rwsu No, please dont put that code in `basic` button. Its the most generic class and all other buttons inherit from it. Only code which is common  for all buttons should be there. Code you add here is pretty specific, so i suggest you make another class, something like `HostPowerButtons` put your code there and change class in toolbars. 

If you want to show those buttons `visible?` method should return true.